src/main/tachyon/tachyon-client.ts

```diff
@@ -45,6 +50,17 @@ export class TachyonClient {
             });
             this.socket.on("unexpected-response", async (req, res) => {
                 res.on("data", (chunk: Buffer) => {
+                    let details: TachyonError | undefined;
+                    try {
+                        details = JSON.parse(chunk.toString());
+                    } catch {
+                        //eat json parsing errors
+                    }
```

Comment:
This parsing is buggy because it reads only a single chunk from response, not whole response. It was maybe sub-optimal but a bit acceptable for stringification below, but when parsing, not really.